solution-jivanov/lib/base.js

```diff
@@ -0,0 +1,45 @@
+// base.js
+// Base class - useful, common routines
+
+'use strict';
+
+class Base {
+    constructor() {
+    }
+
+    static log(...args)
+    {
+        console.log(args);
+    }
+
+    // returns true if n is numeric, false otherwise
+    //based on: http://stackoverflow.com/questions/9716468/is-there-any-function-like-isnumeric-in-javascript-to-validate-numbers
+    static isNumeric(n) {
+        return !isNaN(parseFloat(n)) && isFinite(n);
+    }
+
+    // returns true if s is a string, false otherwise
+    // based on: https://stackoverflow.com/questions/4059147/check-if-a-variable-is-a-string (3rd answer)
+    static isString(s) {
+        return Object.prototype.toString.call(s) === "[object String]";
+    }
+
+    // returns true if f is a function, false otherwise
+    static isFunction(f) {

```

Comment:
Can you help me understand the value of writing low-level operations like `isNumeric`, `isString`, and `isFunction` into the base class?

And are there any alternative methods that you know of to remove this level of abstraction and implementation?

Reply:
There's no value in terms of the particular service, other than me trying to showcase understanding of (a bit) more complicated architectures, abstraction, inheritance, etc. :)

It is also a good point you make about the "...*low-level operations*..."...Unfortunately still not available as built-in features of the language...We should probably submit a proposal for the ESNext...

"*And are there any alternative methods that you know of to remove this level of abstraction and implementation?*"
Certainly, quite a bit of libraries have those built-in (lodash, etc.). ValidateJS does too. We could use it straight in the users model (where we load it anyways). Again, no real need of having them here...